src/plugin/objectSupport/index.js

```diff
@@ -0,0 +1,63 @@
+export default (o, c) => {
+  const proto = c.prototype
+  const isObject = obj => !(obj instanceof Date) && !(obj instanceof Array) && obj instanceof Object
+  const prettyUnit = (u) => {
+    const unit = proto.$utils().p(u)
+    return unit === 'date' ? 'day' : unit
+  }
+  const parseDate = (cfg) => {
+    const { date, utc } = cfg
+    const $d = {}
+    if (isObject(date)) {
+      Object.keys(date).forEach((k) => {
+        $d[prettyUnit(k)] = date[k]
+      })
+      const y = $d.year || 1970
+      const M = $d.month - 1 || 0
+      const d = $d.day || 1
+      const h = $d.hour || 0
+      const m = $d.minute || 0
+      const s = $d.second || 0
+      const ms = $d.millisecond || 0
+      if (utc) {
+        return new Date(Date.UTC(y, M, d, h, m, s, ms))
+      }
+      return new Date(y, M, d, h, m, s, ms)
+    }
+    return date
+  }
+
+  const oldParse = proto.parse
+  proto.parse = function (cfg) {
+    cfg.date = parseDate.bind(this)(cfg)
+    oldParse.bind(this)(cfg)
+  }
+
+  const oldSet = function (int, string) {
+    return this.clone().$set(string, int)
```

Comment:
why a copy-paste instead of `proto.set`

Reply:
here I've changed the `set ` to have the same parameter location as `add`, otherwise you have to judge in `callObject`,  which is inefficient and increases the package size.
or set and add can't share a `callObject`.
---------------------------
also, my attempt to use `proto.set` in `oldSet` will cause a dead loop because `proto.set` is redirected.
Using 'proto.$set' has the problem of the 'this' scope.

Reply:
Oh I see, why useing `return proto.set`  will cause a dead loop?

Reply:
```javasctip
 const oldSet = function (int, string) {
    return proto.set(string, int)
  }
...
proto.set = function (string, int) {
...
}
```
his code causes `oldSet` to enter the newly defined `proto.set` each time, creating A dead loop.

In the new code, I direct used a callback to solve `this` pointing problem.